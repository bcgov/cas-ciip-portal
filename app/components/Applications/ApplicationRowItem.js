import React, {Component} from 'react';
import propTypes from 'prop-types';
import {Button, Badge} from 'react-bootstrap';

class ApplicationRowItem extends Component {
  static propTypes = {
    application: propTypes.shape({
      applicationId: propTypes.number,
      applicationStatus: propTypes.string,
      submissionDate: propTypes.string,
      facilityName: propTypes.string,
      operatorName: propTypes.string,
      reportingYear: propTypes.string,
      bcghgid: propTypes.string
    }).isRequired
  };

  render() {
    const {application} = this.props;
    console.log('my application', application);
    const statusBadgeColor = {
      attention: 'warning',
      pending: 'info',
      declined: 'danger',
      approved: 'success'
    };
    // Const url = `http://localhost:3000/public/dashboard/985719f1-7eae-4c49-88a9-7d6c8edc1ad4?application_id=${application.applicationId}`;
    // Const url = `https://metabase-wksv3k-dev.pathfinder.gov.bc.ca/public/dashboard/bb6a4b75-3a7f-4fab-9268-cb013ecfcb7b?application_id=${application.applicationId}`;
    const applicationDetails = `/application-details?application_id=${application.applicationId}&reportingyear=${application.reportingYear}&bcghgid=${application.bcghgid}`;

    return (
      <tr>
        <td>{application.applicationId}</td>
        <td>{application.operatorName}</td>
        <td>{application.facilityName}</td>
        <td>{application.submissionDate}</td>
        <td>
          <Badge
            pill
            style={{width: '100%'}}
            variant={statusBadgeColor[application.applicationStatus]}
          >
            {application.applicationStatus}
          </Badge>
        </td>
        <td>
          <Button href={applicationDetails} target="_blank" variant="primary">
            View Application
          </Button>
        </td>
      </tr>
    );
  }
}

export default ApplicationRowItem;
