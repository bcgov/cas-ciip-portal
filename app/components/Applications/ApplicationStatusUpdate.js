import React, {Component} from 'react';
import propTypes from 'prop-types';


class ApplicationStatusUpdate extends Component {
  static propTypes = {
    applicationStatus: propTypes.string.isRequired,
    setApplicationStatus: propTypes.func.isRequired
  };

  render() {
    const {applicationStatus, setApplicationStatus} = this.props;
    const statusBadgeColor = {
      pending: 'info',
      attention: 'warning',
      declined: 'danger',
      approved: 'success'
    };

    
}

export default ApplicationStatusUpdate;
