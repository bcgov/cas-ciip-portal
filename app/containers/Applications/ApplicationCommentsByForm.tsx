import React from 'react';
import {createFragmentContainer, graphql, RelayProp} from 'react-relay';
import moment from 'moment';

/*
 * The ApplicationComments renders all the comments on the various sections of the application
 */

interface Props {
  applicationReview;
  relay: RelayProp;
}

export const ApplicationCommentsByForm: React.FunctionComponent<Props> = props => {
  const {applicationReview} = props;
  const reviewComments = applicationReview
    ? applicationReview.reviewCommentsByApplicationReviewId.edges
    : null;

  return (
    <>
      <div key={applicationReview.id} className="review-box">
        <div>
          Set to <strong>{applicationReview.reviewStatus}</strong>
          &nbsp;
          <small>
            {moment(applicationReview.createdAt).format(
              'MMM Do YYYY, h:mm:ss a'
            )}
          </small>
        </div>
        <ul className="comment-box">
          {reviewComments.map(({node}) => (
            <li key={node.id} className="comments">
              <div>{node.description}</div>
              <div>
                <small>
                  {moment(node.createdAt).format('MMM Do YYYY, h:mm:ss a')}
                </small>
              </div>
            </li>
          ))}
        </ul>
      </div>

      <style jsx>{`
        .form-result-box .review-box {
          border-bottom: 1px solid #d2d2d2;
          margin-bottom: 20px;
        }
      `}</style>
    </>
  );
};

export default createFragmentContainer(ApplicationCommentsByForm, {
  applicationReview: graphql`
    fragment ApplicationCommentsByForm_applicationReview on ApplicationReview {
      id
      reviewStatus
      formResultId
      createdAt
      reviewCommentsByApplicationReviewId(first: 2147483647)
        @connection(
          key: "ApplicationCommentsByForm_reviewCommentsByApplicationReviewId"
        ) {
        edges {
          node {
            id
            description
            createdAt
          }
        }
      }
    }
  `
});
