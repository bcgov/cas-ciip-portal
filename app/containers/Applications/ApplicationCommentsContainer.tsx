import React, {useState} from 'react';
import {Collapse, Table, Button} from 'react-bootstrap';
import {createFragmentContainer, graphql, RelayProp} from 'react-relay';
import {ApplicationCommentsContainer_formResult} from 'ApplicationCommentsContainer_formResult.graphql';
import {FontAwesomeIcon} from '@fortawesome/react-fontawesome';
import {faChevronDown, faChevronRight} from '@fortawesome/free-solid-svg-icons';
import ApplicationCommentsBox from './ApplicationCommentsByForm';

/*
 * The ApplicationComments renders all the comments on the various sections of the application
 */

interface Props {
  formResult: ApplicationCommentsContainer_formResult;
  relay: RelayProp;
  review: boolean;
}

export const ApplicationCommentsComponent: React.FunctionComponent<Props> = (
  props
) => {
  const {formResult, review} = props;
  const [isOpen, setIsOpen] = useState(false);
  const [showResolved, toggleShowResolved] = useState(false);
  return (
    <>
      <div key={formResult.id} className="form-result-box">
        <h3>
          {formResult.formJsonByFormId.name}{' '}
          <i
            aria-label="Collapse or expand comment"
            title="Collapse or expand comment"
            style={{float: 'right', marginRight: '10px'}}
            onClick={() => setIsOpen(!isOpen)}
          >
            {isOpen ? (
              <FontAwesomeIcon icon={faChevronRight} />
            ) : (
              <FontAwesomeIcon icon={faChevronDown} />
            )}
          </i>
        </h3>
        <Collapse in={!isOpen}>
          <div>
            <Table hover>
              <thead className="comments-header">
                <tr>
                  <th>General Comment</th>
                  {review ? (
                    <th style={{textAlign: 'right'}}>Resolve/Delete</th>
                  ) : null}
                </tr>
              </thead>
              <tbody>
                {formResult.internalGeneralComments.edges.map(({node}) => {
                  if (node.resolved && !showResolved) return null;
                  if (
                    !review &&
                    formResult.applicationByApplicationId
                      .applicationRevisionStatus.applicationRevisionStatus ===
                      'SUBMITTED'
                  )
                    return null;
                  return (
                    <ApplicationCommentsBox
                      key={node.id}
                      review={review}
                      reviewComment={node}
                      version={formResult.versionNumber}
                      formResultId={formResult.id}
                    />
                  );
                })}
              </tbody>
            </Table>
            <Table hover>
              <thead className="comments-header">
                <tr>
                  <th>Request Changes</th>
                  {review ? (
                    <th style={{textAlign: 'right'}}>Resolve/Delete</th>
                  ) : null}
                </tr>
              </thead>
              <tbody>
                {formResult.requestedChangeComments.edges.map(({node}) => {
                  if (node.resolved && !showResolved) return null;
                  if (
                    !review &&
                    formResult.applicationByApplicationId
                      .applicationRevisionStatus.applicationRevisionStatus ===
                      'SUBMITTED'
                  )
                    return null;
                  return (
                    <ApplicationCommentsBox
                      key={node.id}
                      review={review}
                      reviewComment={node}
                      version={formResult.versionNumber}
                      formResultId={formResult.id}
                    />
                  );
                })}
              </tbody>
            </Table>
            <Button
              variant="link"
              style={{textDecoration: 'none'}}
              onClick={() => toggleShowResolved(!showResolved)}
            >
              {showResolved ? 'Hide Resolved' : 'Show Resolved'}
            </Button>
          </div>
        </Collapse>
      </div>
      <style jsx global>{`
        .form-result-box {
          padding: 25px;
          margin-bottom: 20px;
          border: 1px solid #ff713a;
          border-radius: 5px;
          background-color: #fff4dcab;
          font-size: 14px;
        }

        .form-result-box h3 {
          border-bottom: 1px solid #888;
          padding-bottom: 10px;
          margin-bottom: 20px;
          font-size: 1.25rem;
        }

        .hide-title label.form-label {
          display: none;
        }
        i {
          cursor: pointer;
        }
        .comments-header {
          font-weight: bold;
          color: #036;
        }
        .comments-header th {
          border: 0;
        }
      `}</style>
    </>
  );
};

export default createFragmentContainer(ApplicationCommentsComponent, {
  formResult: graphql`
    fragment ApplicationCommentsContainer_formResult on FormResult {
      id
      versionNumber
      applicationByApplicationId {
        id
        rowId
        applicationRevisionStatus(versionNumberInput: $version) {
          applicationRevisionStatus
        }
      }
      formJsonByFormId {
        rowId
        name
      }
      internalGeneralComments(first: 2147483647)
        @connection(
          key: "ApplicationCommentsContainer_internalGeneralComments"
        ) {
        edges {
          node {
            id
            resolved
            ...ApplicationCommentsByForm_reviewComment
          }
        }
      }
      requestedChangeComments(first: 2147483647)
        @connection(
          key: "ApplicationCommentsContainer_requestedChangeComments"
        ) {
        edges {
          node {
            id
            resolved
            ...ApplicationCommentsByForm_reviewComment
          }
        }
      }
    }
  `
});
