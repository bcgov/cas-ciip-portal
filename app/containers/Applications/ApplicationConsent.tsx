import React from 'react';
import {createFragmentContainer, graphql, RelayProp} from 'react-relay';
import {useRouter} from 'next/router';
import {Button} from 'react-bootstrap';
import updateApplicationRevisionMutation from 'mutations/application/updateApplicationRevisionMutation';
import {ApplicationConsent_application} from 'ApplicationConsent_application.graphql';

interface Props {
  application: ApplicationConsent_application;
  relay: RelayProp;
}

export const ApplicationConsent: React.FunctionComponent<Props> = (props) => {
  const router = useRouter();

  const handleContinueClick = async () => {
    const {environment} = props.relay;

    const variables = {
      input: {
        id: props.application.latestDraftRevision.id,
        applicationRevisionPatch: {
          legalDisclaimerAccepted: true,
          versionNumber: props.application.latestDraftRevision.versionNumber
        }
      }
    };

    await updateApplicationRevisionMutation(environment, variables);

    router.push({
      pathname: '/reporter/application',
      query: {
        applicationId: props.application.id,
        version: props.application.latestDraftRevision.versionNumber
      }
    });
  };

  return (
    <Button variant="primary" size="lg" onClick={handleContinueClick}>
      Consent and continue
    </Button>
  );
};

export default createFragmentContainer(ApplicationConsent, {
  application: graphql`
    fragment ApplicationConsent_application on Application {
      id
      latestDraftRevision {
        id
        versionNumber
      }
    }
  `
});
