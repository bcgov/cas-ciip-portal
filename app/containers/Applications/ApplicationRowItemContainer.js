import React from 'react';
import {Button, Badge} from 'react-bootstrap';
import {graphql, createFragmentContainer} from 'react-relay';
import Link from 'next/link';

const ApplicationRowItemContainer = props => {
  const {ciipApplication} = props;
  const statusBadgeColor = {
    attention: 'warning',
    pending: 'info',
    declined: 'danger',
    approved: 'success'
  };

  return (
    <tr>
      <td>{ciipApplication.rowId}</td>
      <td>{ciipApplication.operatorName}</td>
      <td>{ciipApplication.facilityName}</td>
      <td>{ciipApplication.submissionDate}</td>
      <td>
        <Badge
          pill
          style={{width: '100%'}}
          variant={statusBadgeColor[ciipApplication.applicationStatus]}
        >
          {ciipApplication.applicationStatus}
        </Badge>
      </td>
      <td>
        <Link
          href={{
            pathname: '/applicationdetails',
            query: {
              applicationId: ciipApplication.rowId,
              reportingYear: ciipApplication.reportingYear,
              bcghgid: ciipApplication.bcghgid
            }
          }}
        >
          <Button variant="primary">View Application</Button>
        </Link>
      </td>
    </tr>
  );
};

export default createFragmentContainer(ApplicationRowItemContainer, {
  ciipApplication: graphql`
    fragment ApplicationRowItemContainer_ciipApplication on CiipApplication {
      rowId
      facilityName
      operatorName
      applicationStatus
      submissionDate
      reportingYear
      bcghgid
    }
  `
});
