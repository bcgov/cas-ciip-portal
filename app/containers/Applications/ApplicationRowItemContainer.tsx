import React from 'react';
import {Button, Badge} from 'react-bootstrap';
import {graphql, createFragmentContainer} from 'react-relay';
import Link from 'next/link';

export const ApplicationRowItem = props => {
  const {ciipApplication = {}} = props;
  const statusBadgeColor = {
    DRAFT: 'warning',
    PENDING: 'info',
    REJECTED: 'danger',
    APPROVED: 'success'
  };

  return (
    <tr>
      <td>{ciipApplication.rowId}</td>
      <td>{ciipApplication.operatorName}</td>
      <td>{ciipApplication.facilityName}</td>
      <td>{ciipApplication.reportingYear}</td>
      <td>{ciipApplication.submissionDate}</td>
      <td>
        <Badge
          pill
          style={{width: '100%'}}
          variant={statusBadgeColor[ciipApplication.applicationStatus]}
        >
          {ciipApplication.applicationStatus}
        </Badge>
      </td>
      <td>
        <Link
          href={{
            pathname: '/application-review',
            query: {
              applicationId: ciipApplication.rowId,
              applicationGUID: ciipApplication.applicationByRowId.id,
              reportingYear: ciipApplication.reportingYear,
              bcghgid: ciipApplication.bcghgid
            }
          }}
        >
          <Button variant="primary">View Application</Button>
        </Link>
      </td>
    </tr>
  );
};

export default createFragmentContainer(ApplicationRowItem, {
  ciipApplication: graphql`
    fragment ApplicationRowItemContainer_ciipApplication on CiipApplication {
      id
      rowId
      facilityName
      operatorName
      submissionDate
      reportingYear
      bcghgid
      applicationStatus
      applicationByRowId {
        id
      }
    }
  `
});
