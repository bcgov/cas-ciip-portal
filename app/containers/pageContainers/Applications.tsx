import React, {Component} from 'react';
import {graphql} from 'react-relay';
import {ApplicationsQueryResponse} from '__generated__/ApplicationsQuery.graphql';
import DefaultLayout from '../../layouts/default-layout';
import ApplicationListContainer from '../Applications/ApplicationListContainer';

interface Props {
  query?: ApplicationsQueryResponse['query'];
}

class Applications extends Component<Props> {
  // In downstream component
  // onClick={(event) => handleApplicationsAction({action: Applications.SORT_ACTION})})
  // I didn't fully understand how to make this work, so I moved on with the way I did.
  static query = graphql`
    query ApplicationsQuery(
      $orderByField: String
      $direction: String
      $searchField: String
      $searchValue: String
    ) {
      query {
        session {
          ...Header_session
        }
        ...ApplicationListContainer_query
          @arguments(
            orderByField: $orderByField
            direction: $direction
            searchField: $searchField
            searchValue: $searchValue
          )
      }
    }
  `;

  state = {
    orderByField: 'operator_name',
    direction: 'ASC',
    orderByDisplay: 'Operator Name',
    searchField: null,
    searchValue: null,
    searchDisplay: 'No Filter'
  };

  static async getInitialProps() {
    return {
      variables: {
        orderByField: 'operator_name',
        direction: 'ASC',
        searchField: null,
        searchValue: null
      }
    };
  }

  toggleDirection = () => {
    this.state.direction === 'ASC'
      ? this.setState({direction: 'DESC'})
      : this.setState({direction: 'ASC'});
  };

  sortApplications = (event, eventKey) => {
    this.setState({
      orderByField: eventKey,
      orderByDisplay: event.target.text
    });
  };

  applySearchField = (event, eventKey) => {
    this.setState({
      searchField: eventKey,
      searchDisplay: event.target.text,
      searchValue: null
    });
  };

  applySearchValue = event => {
    if (this.state.searchField !== 'none') {
      this.setState({searchValue: event.nativeEvent.target[0].value});
    }
  };

  handleEvent = (event, eventKey) => {
    event.preventDefault();
    event.stopPropagation();
    event.persist();
    this[event.target.id](event, eventKey);
  };

  render() {
    const {query} = this.props;
    return (
      <>
        <DefaultLayout title="Applications" session={query.session}>
          <ApplicationListContainer
            query={query}
            orderByDisplay={this.state.orderByDisplay}
            searchDisplay={this.state.searchDisplay}
            direction={this.state.direction}
            orderByField={this.state.orderByField}
            searchField={this.state.searchField}
            searchValue={this.state.searchValue}
            handleEvent={this.handleEvent}
          />
        </DefaultLayout>
      </>
    );
  }
}

export default Applications;
