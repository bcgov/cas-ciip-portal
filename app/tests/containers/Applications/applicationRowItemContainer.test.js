import React from 'react';
import {shallow} from 'enzyme';
import {ApplicationRowItem} from '../../../containers/Applications/ApplicationRowItemContainer';

describe('ApplicationRowItem', () => {
  it('should render the application', () => {
    const ciipApplication = {
      rowId: '9',
      applicationByRowId: {id: 'ABC'},
      applicationStatus: 'pending',
      facilityName: 'facility1',
      operatorName: 'operator1',
      submissionDate: 'Sun, 17 Dec 1995 03:24:00 GMT'
    };
    const render = shallow(
      <ApplicationRowItem ciipApplication={ciipApplication} />
    );
    expect(render).toMatchSnapshot();
  });
});
