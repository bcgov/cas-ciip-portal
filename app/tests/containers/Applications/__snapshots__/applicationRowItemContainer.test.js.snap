// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ApplicationRowItemContainer should render the application 1`] = `
<tr>
  <td />
  <td>
    operator1
  </td>
  <td>
    facility1
  </td>
  <td>
    Sun, 17 Dec 1995 03:24:00 GMT
  </td>
  <td>
    <Badge
      pill={true}
      style={
        Object {
          "width": "100%",
        }
      }
      variant="info"
    >
      pending
    </Badge>
  </td>
  <td>
    <Link
      href={
        Object {
          "pathname": "/application-details",
          "query": Object {
            "applicationId": undefined,
            "bcghgid": undefined,
            "reportingYear": undefined,
          },
        }
      }
    >
      <Button
        active={false}
        disabled={false}
        type="button"
        variant="primary"
      >
        View Application
      </Button>
    </Link>
  </td>
</tr>
`;
